module/zfs/zfs_znode.c

```diff
@@ -526,26 +525,12 @@ zfs_inode_update_impl(znode_t *zp, boolean_t new)
 	if (zfsctl_is_node(ip))
 		return;
 
-	sa_lookup(zp->z_sa_hdl, SA_ZPL_ATIME(zsb), &atime, 16);
-	sa_lookup(zp->z_sa_hdl, SA_ZPL_MTIME(zsb), &mtime, 16);
-	sa_lookup(zp->z_sa_hdl, SA_ZPL_CTIME(zsb), &ctime, 16);
-
 	dmu_object_size_from_db(sa_get_db(zp->z_sa_hdl), &blksize, &i_blocks);
 
 	spin_lock(&ip->i_lock);
 	ip->i_mode = zp->z_mode;
 	zfs_set_inode_flags(zp, ip);
 	ip->i_blocks = i_blocks;
-
-	/*
-	 * Only read atime from SA if we are newly created inode (or rezget),
-	 * otherwise i_atime might be dirty.
-	 */
-	if (new)
```

Comment:
The existing range lock will guarantee a consistent ordering as long as you perform the inode updates while it's being held.  Which is why `zfs_inode_update()` always happens before `zfs_range_unlock()`.